eternalcore-api/src/main/java/com/eternalcode/core/feature/jail/event/JailDetainEvent.java

```diff
@@ -0,0 +1,44 @@
+package com.eternalcode.core.feature.jail.event;
+
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+
+public class JailDetainEvent extends PlayerEvent implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final CommandSender detainedBy;
+
+    private boolean cancelled = false;
```

Comment:
The primitive logical value has a default value of false

```suggestion
    private boolean cancelled;
```